File: sdk/typescript/examples/langchain_integration.ts
import { memoryOSApiKey, memoryOSBaseUrl } from "./env.js";
import { MemoryOS } from "../src/index.js";

async function main(): Promise<void> {
  const client = new MemoryOS(memoryOSApiKey(), memoryOSBaseUrl());
  const results = await client.get({
    query: "programming language preferences",
    externalUserId: "student_44821",
    limit: 3,
    timeFilterDays: 30,
  });
  const context = results.hasContext ? results.systemPromptAddition : "";
  console.log("Inject this into your LangChain prompt:");
  console.log(context);
}

void main();

File: sdk/python/examples/langchain_integration.py
import os

from memoryos import Memory


def main() -> None:
    memory = Memory(
        api_key=os.environ["MEMORYOS_API_KEY"],
        base_url=os.getenv("MEMORYOS_BASE_URL", "http://127.0.0.1:8000"),
    )
    results = memory.get(
        query="programming language preferences",
        external_user_id="student_44821",
        limit=3,
        time_filter_days=30,
    )
    context = results.system_prompt_addition if results.has_context else ""
    print("Inject this into your LangChain prompt:")
    print(context)


if __name__ == "__main__":
    main()
